test/types/index.test-d.ts

```diff
@@ -65,7 +66,7 @@ const augmentedRouteOptions: RouteOptions = {
   },
   wsHandler: (connection, request) => {
     expectType<SocketStream>(connection);
-    expectType<FastifyRequest<RequestGenericInterface>>(request)
+    expectType<FastifyRequest<RouteGenericInterface>>(request)
   },
 };
```

Comment:
Want to add a test for the new functionality based on your original issue? Thanks for the contribution!

Reply:
I never really tested "types", didn't know that was a thing until now :D

Reply:
So... what next then? :D

Reply:
To add a test case you'd need to copy one of these stanzas and exercise specifically the new types you added! This file is using `tsd` which is a typescript testing tool and you can find more info in its docs.